### PR TITLE
OH2-361 | Upgrade to react 18

### PR DESCRIPTION
**Pull request by SteveGT96**
Includes:

- React 18.3.1
- mui 5.16.4

See [OH2-361](https://openhospital.atlassian.net/browse/OH2-361)

[OH2-361]: https://openhospital.atlassian.net/browse/OH2-361?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cypress/integrations/admin_activities/operations_activities/edit_operation_activity.cy.ts [hunk 1]
```diff
@@ -25,7 +25,8 @@ describe("Operations Edit Activity specs", () => {
   it("should successfully save operation changes", () => {
     cy.byId("type").click();
     cy.get('.MuiAutocomplete-popper li[data-option-index="0"]').click();
-    cy.dataCy("submit-form").click();
+    cy.byId("type").focus();
+    cy.dataCy("submit-form").click().click();
```
Comment (gasp) [inline cypress/integrations/admin_activities/operations_activities/edit_operation_activity.cy.ts:29]: this looks strange, is it standard behaviour?

Comment (SteveGT96) [reply cypress/integrations/admin_activities/operations_activities/edit_operation_activity.cy.ts:29]: > this looks strange, is it standard behaviour?

Really strange. After upgrate to react 18, some time clicking a second or third on the a  button doesn't complete the click event. This tweak fixes the problem.

Comment (gasp) [reply cypress/integrations/admin_activities/operations_activities/edit_operation_activity.cy.ts:29]: have you tried `.click({ force: true })` ?

**Comment by SilverD3** [comment]
The design of the filter in the dashboard isn't aligned.
Screenshot 2024-07-31 at 14-52-40 Open Hospital
Before
Screenshot 2024-07-31 at 14-53-39 Open Hospital
After

**Review by SilverD3 — COMMENTED** [review]
Apart from the design problem I mentioned in my comment, I think everything's okay. I've gone through several pages and apart from the spaces between elements which have increased a little (which isn't easily noticeable), there's nothing else to report.

**Review by mwithi — APPROVED** [review]
Tested and LGTM too. Thanks a lot!

**Pull request merged**